src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebRequestPSCmdlet.CoreClr.cs

```diff
@@ -347,14 +368,29 @@ protected override void ProcessRecord()
                         WriteVerbose(reqVerboseMsg);
 
                         HttpResponseMessage response = GetResponse(client, request);
-                        response.EnsureSuccessStatusCode();
 
                         string contentType = ContentHelper.GetContentType(response);
                         string respVerboseMsg = string.Format(CultureInfo.CurrentCulture,
                             "received {0}-byte response of content type {1}",
                             response.Content.Headers.ContentLength,
                             contentType);
                         WriteVerbose(respVerboseMsg);
+
+                        if (!response.IsSuccessStatusCode)
+                        {
+                            string message = String.Format(CultureInfo.CurrentCulture, WebCmdletStrings.ResponseStatusCodeFailure,
+                                Convert.ToInt32(response.StatusCode).ToString(), response.ReasonPhrase);
+                            HttpResponseException httpEx = new HttpResponseException(message, response);
+                            ErrorRecord er = new ErrorRecord(httpEx, "WebCmdletWebResponseException", ErrorCategory.InvalidOperation, request);
+                            string detailMsg = "";
+                            using (StreamReader reader = new StreamReader(StreamHelper.GetResponseStream(response)))
```

Comment:
Exceptions may be thrown out in `StreamHelper.GetResponseStream(response)` or `reader.ReadToEnd()`. We should put the `using` block in a try/catch-all block, just like [what we did](https://github.com/PowerShell/PowerShell/blob/master/src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/FullClr/WebRequestPSCmdlet.FullClr.cs#L473) in full ps.